File: sudoq-app/sudoqmodel/src/test/java/de/sudoq/model/sudoku/complexity/ComplexityTests.java
package de.sudoq.model.sudoku.complexity;

import static org.junit.Assert.assertTrue;

import org.junit.Test;

public class ComplexityTests {

	@Test
	public void test() {
		Complexity[] types = Complexity.values();
		for (Complexity type : types) {
			assertTrue(Complexity.valueOf(type.toString()).equals(type));
		}
	}

}

File: sudoq-app/sudoqmodel/src/test/kotlin/de/sudoq/model/sudoku/complexity/ComplexityTests.kt
package de.sudoq.model.sudoku.complexity

import org.amshove.kluent.`should be equal to`
import org.junit.jupiter.params.ParameterizedTest
import org.junit.jupiter.params.provider.EnumSource

class ComplexityTests {

    /**
     * Verifies that toString hasn't been tampered with.
     * Dubious use.
     */
    @ParameterizedTest
    @EnumSource(Complexity::class)
    fun test(c: Complexity) {
        Complexity.valueOf(c.toString()).`should be equal to` (c)
    }
}